src/org/sosy_lab/java_smt/test/ModelTest.java

```diff
@@ -2254,6 +2259,312 @@ public void arrayTest5()
     checkModelIteration(formula, false);
   }
 
+  /** Returns the arguments of an UF or array definition from the model. */
+  private Iterable<Formula> getFunctionArgs(Formula pValue) {
+    requireVisitor();
+    return mgr.visit(
+        pValue,
+        new DefaultFormulaVisitor<>() {
+          @Override
+          protected Iterable<Formula> visitDefault(Formula f) {
+            throw new IllegalArgumentException();
+          }
+
+          @Override
+          public Iterable<Formula> visitFreeVariable(Formula f, String name) {
+            // Base case for array values
+            return ImmutableList.of();
+          }
+
+          @Override
+          public Iterable<Formula> visitFunction(
+              Formula f, List<Formula> args, FunctionDeclaration<?> functionDeclaration) {
+            if (functionDeclaration.getKind().equals(FunctionDeclarationKind.SELECT)) {
+              return FluentIterable.concat(
+                  getFunctionArgs(args.get(0)), ImmutableList.of(args.get(1)));
+            } else {
+              // Assume its an UF value
```

Comment:
`visitFunction()` is defined to be called on any function, including UFs. I would recommend explicitly checking for UFs. You could take a look at the `VariableAndUFExtractor` for inspiration.